supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
actually, since `s.tasks` is still buffered, we may need to have `SendTask` accept the `ctx` and then (in `SendTask(ctx, e)`) something like:
```go
for {
    select {
    case ctx.Done():
        // cancel send task and return error in evt.ErrChan()
        // no need to add the task anymore as it's being canceled by the caller
    case s.tasks <- evt:
        TaskCounter.Add(1)
    default:
        continue // retry
    }
}
```